src/server.py

```diff
@@ -257,26 +266,35 @@ def _ensure_release_countdown(self):
         self._countdown_id = GLib.timeout_add_seconds(
             self._TIMEOUT_S, self.release, priority=GLib.PRIORITY_LOW)
 
-    def play_sound(self, sound_event_id, connection, sender, path, iface,
-                   invocation):
+    def _play_sound(self, sound_event_id, connection, sender, path, iface,
+                    invocation, options=None):
         if sound_event_id not in self.metadata:
             invocation.return_dbus_error(
                 self._DBUS_UNKNOWN_SOUND_EVENT_ID,
                 "sound event with id %s does not exist" % sound_event_id)
             return
-
         self._cancel_countdown()
         self.hold()
 
         uuid_ = str(uuid.uuid4())
-        self.players[uuid_] = HackSoundPlayer(self.metadata[sound_event_id],
-                                              sender)
+        metadata = self.metadata[sound_event_id]
+        self.players[uuid_] = HackSoundPlayer(metadata, sender, options)
         self.players[uuid_].connect("eos", self.__player_eos_cb, uuid_)
         self.players[uuid_].connect("error", self.__player_error_cb, uuid_,
                                     connection, path, iface)
         self.players[uuid_].play()
         invocation.return_value(GLib.Variant('(s)', (uuid_, )))
 
+    def play_sound(self, sound_event_id, connection, sender, path, iface,
```

Comment:
I would remove these and just call `self.play_sound()` directly in both the `PlaySound` and `PlayFull` cases below.